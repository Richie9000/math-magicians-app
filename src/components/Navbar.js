import React from 'react';

const Navbar = () => (
  <nav>
    <ul>
      <li>Home</li>
      <li>Calculator</li>
      <li>Quotes</li>
      <li id="title">Math Magicians</li>
    </ul>
  </nav>
);

export default Navbar;
