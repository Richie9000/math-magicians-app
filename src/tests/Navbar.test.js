import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';
import '@testing-library/jest-dom';

describe('App', () => {
  test('renders "Home" link in Navbar component', () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(screen.getByText('Home')).toBeInTheDocument();
  });

  test('renders "Calculator" link in Navbar component', () => {
    render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(screen.getByText('Calculator')).toBeInTheDocument();
  });

  test('Navbar snapshot test', () => {
    const myrender = render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(myrender).toMatchSnapshot();
  });
});
