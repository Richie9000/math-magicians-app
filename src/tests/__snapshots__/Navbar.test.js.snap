// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Navbar snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <ul>
          <li
            id="title"
          >
            <a
              class="links"
              href="/"
            >
              Math Magicians
            </a>
          </li>
          <li>
            <a
              class="links"
              href="/quote"
            >
              Quote
            </a>
          </li>
          <li>
            <a
              class="links"
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li>
            <a
              class="links"
              href="/"
            >
              Home
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <ul>
        <li
          id="title"
        >
          <a
            class="links"
            href="/"
          >
            Math Magicians
          </a>
        </li>
        <li>
          <a
            class="links"
            href="/quote"
          >
            Quote
          </a>
        </li>
        <li>
          <a
            class="links"
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            class="links"
            href="/"
          >
            Home
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
